neural/neural_discern_run_script.py

```diff
@@ -0,0 +1,350 @@
+        'max_folds': 5,
+        'num_epochs': 25,
+        'verbose': True,
+        'experiment_to_rerun': None,
+        'questions': (4, 5, 9, 10, 11),
+        'question_gpu_map': {4: 1, 5: 2, 9: 3, 10: 4, 11: 5},
+        'base_dir': args.base_dir,
+    }
+
+    if config['hyperparam_search_dir'] and config['run_hyper_param_search']:
+        print("WARNING: you selected a hyperparam search dir while also setting run-hyper-param-search as True. "
+              "The pre-built hyperparam search dir will be used. Hyperparam search will not be run.")
+
+    # under test mode (for faster debugging), run a smaller set of partitions and epochs, and no hyper param search
+    if config['test_mode']:
+        config['max_folds'] = 2  # max number of data partition folds to run (for faster testing)
+        config['num_epochs'] = 2
+        config['run_hyper_param_search'] = False
+
+    if config['experiment_to_rerun'] is None:
+        time_stamp = datetime.datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
+        if config['test_mode']:
+            exp_dir = os.path.join(config['base_dir'], 'experiments', 'tests', time_stamp)
+        else:
+            exp_dir = os.path.join(config['base_dir'], 'experiments', time_stamp)
+        create_directory(exp_dir)
+    else:
+        exp_dir = os.path.join(config['base_dir'], 'experiments', config['experiment_to_rerun'])
+    config['exp_dir'] = exp_dir
+
+    if config['biobert']:
+        config['bert_model_dir'] = os.path.join(config['base_dir'], 'pytorch_biobert')
+        config['sents_embed_dir_name'] = 'sents_bert_embed_cased'
+        config['data_dir'] = os.path.join(config['base_dir'], 'proc_data_cased')
+    else:
+        config['bert_model_dir'] = os.path.join(config['base_dir'], 'aws_downloads/bert-base-uncased.tar.gz')
+        config['sents_embed_dir_name'] = 'sents_bert_embed_uncased'
+        config['data_dir'] = os.path.join(config['base_dir'], 'proc_data_uncased')
+
+    # print config to screen
+    print("{0} RUNNING WITH CONFIG {0}".format('='*10))
+    for c in config:
+        print("{}: {}".format(c, config[c]))
+    print("{0}".format('='*10*4))
+
+    # ---
+
+    verbose = config['verbose']
+
+    # get the gpu index for the first question that is to be run
+    default_gpu_index = config['question_gpu_map'][config['questions_to_run'][0]]
+    default_gpu_index = 0
+    default_device = get_device(to_gpu=False, index=default_gpu_index)
+
+    verbose_print("Loading objects...", verbose)
+    data_dir = config['data_dir']
+    q_partitions, docs_data_tensor, proc_articles_dict, proc_articles_repr, proc_config = read_pickles(data_dir)
+    q_docpartitions = build_doc_partitions(config['questions'], proc_articles_repr, proc_articles_dict, proc_config,
+                                           docs_data_tensor, q_partitions)
+
+    if config['biobert']:
+        pytorch_dump_path = create_directory('pytorch_biobert', config['base_dir'])
```

Comment:
is not this path already available in `config['bert_model_dir']` in line 272?

Reply:
correct!